internal/usecase/repository/repo.go

```diff
@@ -0,0 +1,43 @@
+package repository
+
+import (
+	"address-book/internal/entity"
+)
+
+type Repository []*entity.AddressField
+
+func NewRepository() *Repository {
+	return new(Repository)
+}
+
+func (r Repository) FindField(param string) (index int, ok bool) {
+	for i, field := range r {
+		if field.Name == param {
+			return i, true
+		}
+	}
+	for i, field := range r {
+		if field.Phone == param {
+			return i, true
+		}
+	}
+	for i, field := range r {
+		if field.Phone == param {
+			return i, true
+		}
+	}
+
+	return 0, false
+}
+
+func (r Repository) AddField(name, address, phone string) Repository {
```

Comment:
Let's assume, we have only one user in our address-book: ("John Doe", "+1-23-456-78-90", "Baker street, 10"). Then we try to add the same user into the address book again, will it be added or not? Address-book should not contain a full duplicates, right?